### PR TITLE
fix: sig verifying error handling

### LINKED ISSUE #210: Don't leak signature error during relay verification
https://github.com/ralexstokes/mev-rs/blob/4469676ce10eb023bb5478e88429aaed84fbf619/mev-relay-rs/src/relay.rs#L535

This check was added in #208 but leaks the underlying error to the caller if there is a problem.

For security reasons, it is better to return an opaque "failure" error. Note that we still want to log the error locally. Reference the pattern elsewhere like here: https://github.com/ralexstokes/mev-rs/blob/main/mev-relay-rs/src/relay.rs#L529-L532

Let's update this to check for the `Err` variant, `warn!` about the bad signature, but only return `RelayError::InvalidSignedBlindedBeaconBlock` to the caller

### PR DESCRIPTION
**Pull request by sarmatdev**
Fixes #210 `verify_blinded_block_signature` error handling

### CODE REVIEW

File: mev-relay-rs/src/relay.rs [hunk 1]
```diff
@@ -532,7 +532,10 @@ impl BlindedBlockProvider for Relay {
             }
         }
 
-        verify_blinded_block_signature(&auction_request, signed_block, self)?;
+        if let Err(err) = verify_blinded_block_signature(&auction_request, signed_block, self) {
+            warn!(%err, %auction_request, "invalid incoming signed blinded block signature");
```
Comment (ralexstokes) [inline mev-relay-rs/src/relay.rs:536]:
```suggestion
            warn!(%err, %auction_request, "invalid signature on incoming signed blinded block");
```

File: mev-relay-rs/src/relay.rs [hunk 2]
```diff
@@ -532,7 +532,10 @@ impl BlindedBlockProvider for Relay {
...
+            return Err(RelayError::InvalidSignedBlindedBeaconBlock.into())
+        };
```
Comment (ralexstokes) [inline mev-relay-rs/src/relay.rs:538]:
```suggestion
        }
```

**Review by ralexstokes — APPROVED** [review]
looks great thank you! adding some small nits, and hoping that fixes the CI which seems blocked right now

**Comment by sarmatdev** [comment]
Thank you! I'd like to collaborate on this project. Where and how we can communicate?

**Comment by ralexstokes** [comment]
@sarmatdev on twitter? you can dm me there

**Comment by sarmatdev** [comment]
I don't have premium so I can't dm you there

**Comment by ralexstokes** [comment]
what's your handle? I can try to dm you

**Comment by sarmatdev** [comment]
0xsarmat

**Pull request merged**